### PR TITLE
Segment validation

### PR DESCRIPTION
**Pull request by dylanlott**
Adds segment validation to pointerdb server

---
This change is [Reviewable](https://reviewable.io/reviews/storj/storj/280)

### CODE REVIEW

File: pkg/pointerdb/pointerdb.go [hunk 1]
```diff
@@ -53,11 +59,41 @@ func (s *Server) validateAuth(APIKey []byte) error {
...
 
+func (s *Server) validateSegment(req *pb.PutRequest) error {
+	min := s.config.MinInlineSegmentSize
+	max := s.config.MaxInlineSegmentSize
+	inlineSize := len(req.GetPointer().InlineSegment)
+	pointer := req.GetPointer()
+	remote := pointer.Remote
+
+	fmt.Printf("Pointer is %+v\n", pointer)
+	fmt.Printf("Remote is %+v\n", remote)
```
Comment (jtolds) [inline pkg/pointerdb/pointerdb.go]: can you change these to debug logs?

Comment (dylanlott) [reply pkg/pointerdb/pointerdb.go]: they weren't supposed to be in there, I just removed them instead.

File: pkg/pointerdb/pointerdb.go [hunk 2]
```diff
@@ -24,24 +26,28 @@ import (
 
 var (
 	mon = monkit.Package()
+	segmentError = errors.New("Invalid segment formatting")
```
[the remark this answers is not in the document]
Comment (dylanlott) [reply pkg/pointerdb/pointerdb.go]: Done

File: pkg/pointerdb/pointerdb.go [hunk 3]
```diff
@@ -53,11 +59,41 @@ func (s *Server) validateAuth(APIKey []byte) error {
...
+	pointer := req.GetPointer()
+	remote := pointer.Remote
+
+	fmt.Printf("Pointer is %+v\n", pointer)
+	fmt.Printf("Remote is %+v\n", remote)
+
+	if inlineSize < min {
+		if remote != nil {
+			s.logger.Error("segment validation error: ", zap.Error(segmentError))
+			return segmentError
```
[the remark this answers is not in the document]
Comment (dylanlott) [reply pkg/pointerdb/pointerdb.go]: Done

File: pkg/pointerdb/pointerdb.go [hunk 4]
```diff
@@ -53,11 +59,41 @@ func (s *Server) validateAuth(APIKey []byte) error {
...
+}
+
 // Put formats and hands off a key/value (path/pointer) to be saved to boltdb
 func (s *Server) Put(ctx context.Context, req *pb.PutRequest) (resp *pb.PutResponse, err error) {
 	defer mon.Task()(&ctx)(&err)
 	s.logger.Debug("entering pointerdb put")
 
+	err = s.validateSegment(req)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, err.Error())
```
[the remark this answers is not in the document]
Comment (dylanlott) [reply pkg/pointerdb/pointerdb.go]: Done

File: pkg/pointerdb/pointerdb.go [hunk 5]
```diff
@@ -53,11 +59,41 @@ func (s *Server) validateAuth(APIKey []byte) error {
...
+	inlineSize := len(req.GetPointer().InlineSegment)
+	pointer := req.GetPointer()
+	remote := pointer.Remote
+
+	fmt.Printf("Pointer is %+v\n", pointer)
+	fmt.Printf("Remote is %+v\n", remote)
+
+	if inlineSize < min {
+		if remote != nil {
+			s.logger.Error("segment validation error: ", zap.Error(segmentError))
```
[the remark this answers is not in the document]
Comment (dylanlott) [reply pkg/pointerdb/pointerdb.go]: I think you're right, I decided to remove the logs from here and just return the zeebo error instead.

File: pkg/pointerdb/pointerdb.go [hunk 6]
```diff
@@ -53,11 +58,35 @@ func (s *Server) validateAuth(APIKey []byte) error {
 	return nil
 }
 
+func (s *Server) validateSegment(req *pb.PutRequest) error {
+	min := s.config.MinInlineSegmentSize
+	max := s.config.MaxInlineSegmentSize
+	inlineSize := len(req.GetPointer().InlineSegment)
+	remote := req.GetPointer().Remote
+
+	if inlineSize < min {
```
Comment (jtolds) [inline pkg/pointerdb/pointerdb.go]: wait, i think this should be `if remote != nil && remote.Size < min`. sorry for missing it the first time around

File: pkg/pointerdb/config.go [hunk 7]
```diff
@@ -18,6 +18,8 @@ import (
 // PointerDB responsibility
 type Config struct {
 	DatabaseURL string `help:"the database connection string to use" default:"bolt://$CONFDIR/pointerdb.db"`
+	MinInlineSegmentSize int64 `default:"1240" help:"minimum inline segment size"`
+	MaxInlineSegmentSize int `default:"8000" help:"maximum inline segment size"`
```
Comment (egonelbre) [inline pkg/pointerdb/config.go]: formatting

**Pull request merged**